xbmc/interfaces/builtins/GUIBuiltins.cpp

```diff
@@ -86,13 +89,16 @@ static int ActivateWindow(const std::vector<std::string>& params2)
         bIsSameStartFolder = static_cast<CGUIMediaWindow*>(activeWindow)->IsSameStartFolder(params[0]);
     }
 
-    // let the window know it is being replaced
-    if (Replace)
-      params.emplace_back("replace");
-
     // activate window only if window and path differ from the current active window
     if (iWindow != CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow() || !bIsSameStartFolder)
     {
+      // if the window doesn't change, make sure it knows it's gonna be replaced
+      // this ensures setting the start directory if we switch paths
+      // if we change windows, that's done anyway
+      if (iWindow == CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow() &&
+          !params.empty() && Replace)
```

Comment:
Always evaluate AND-ed terms from "cheap" to "expensive", please. If `Replace` is false there is no need to do the more expensive function call, for example.

=> 

```c++
if (bReplace && !params.empty() && iWindow == CServiceBroker::GetGUI()->GetWindowManager().GetActiveWindow())
```

Reply:
Sure np.